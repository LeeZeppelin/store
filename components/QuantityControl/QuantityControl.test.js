import React from 'react';

import { render, fireEvent } from 'react-testing-library';
import { QuantityControl } from '.';

describe('QuantityControl', () => {
  it('renders correctly ', () => {
    const { asFragment } = render(
      <QuantityControl quantity={1} increment={() => {}} decrement={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the passed in quantity', () => {
    const { getByText } = render(
      <QuantityControl
        quantity={298}
        increment={() => {}}
        decrement={() => {}}
      />
    );
    expect(getByText('298')).toBeInTheDocument();
  });

  it('calls the passed in increment function on + click', () => {
    const mockIncrement = jest.fn();
    const { getByText } = render(
      <QuantityControl
        quantity={1}
        increment={mockIncrement}
        decrement={() => {}}
      />
    );

    fireEvent.click(getByText('+'));
    expect(mockIncrement).toHaveBeenCalled();
  });

  it('calls the passed in decrement function on - click when quantity is above 1', () => {
    const mockDecrement = jest.fn();
    const { getByText } = render(
      <QuantityControl
        quantity={2}
        increment={() => {}}
        decrement={mockDecrement}
      />
    );

    fireEvent.click(getByText('-'));
    expect(mockDecrement).toHaveBeenCalled();
  });

  it('does not call the passed in decrement function on - click when quantity is 1', () => {
    const mockDecrement = jest.fn();
    const { getByText } = render(
      <QuantityControl
        quantity={1}
        increment={() => {}}
        decrement={mockDecrement}
      />
    );

    fireEvent.click(getByText('-'));
    expect(mockDecrement).not.toHaveBeenCalled();
  });
});
