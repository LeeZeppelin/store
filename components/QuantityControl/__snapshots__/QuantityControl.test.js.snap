// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityControl renders correctly  1`] = `
<DocumentFragment>
  <p
    class="QuantityControl__Quantity-sc-7qsiy5-1 cJckkI"
  >
    1
  </p>
  <div
    class="QuantityControl__Controls-sc-7qsiy5-2 ccBOgV"
  >
    <button
      class="QuantityControl__QuantityButton-sc-7qsiy5-0 hQVPGU"
      type="button"
    >
      +
    </button>
    <button
      class="QuantityControl__QuantityButton-sc-7qsiy5-0 hQVPGU"
      disabled=""
      type="button"
    >
      -
    </button>
  </div>
</DocumentFragment>
`;
